source/prefab/Card.h

```diff
@@ -0,0 +1,83 @@
+#define EMP_CARD_INTERFACE_H
+
+#include "../web/Div.h"
+#include "../tools/string_utils.h"
+#include "Collapse.h"
+#include "FontAwesomeIcon.h"
+
+namespace emp {
+    namespace prefab{
+        class Card : public web::Div{
+            private:
+                web::Div collapse_link;
+                web::Div collapse_body;
+                std::string card_base = this->GetID();
+                web::Div card_header{emp::to_string(card_base, "_card_header")};
+                web::Div card_body{emp::to_string(card_base, "_card_body")};
+                void AddBootstrap(){
+                    this->SetAttr("class", "card");
+                    card_header.SetAttr("class", "card-header");
+                    card_body.SetAttr("class", "card-body");
+                    // card_header.AddClass("collapse_toggle_card_header");
+                }
+            public:
+                enum Collapse {NONE, OPEN, CLOSED}; // collapse options for constructing a card 
+                Card(Collapse state=NONE, bool showGlyphs=true, const std::string & in_name=""): web::Div(in_name){
+                    AddBootstrap();
+                    if(state != NONE){ // if card is collapsible, make the collapse link the head of the card
+                        if(state == OPEN){ 
+                            prefab::Collapse accordion(card_header, card_body, true, emp::to_string(card_base+ "_card_collapse"));
+                            collapse_link = accordion.GetLinkDiv();
+                            collapse_body = accordion.GetToggleDiv();
+                        }
+                        else{
+                            prefab::Collapse accordion(card_header, card_body, false, emp::to_string(card_base + "_card_collapse"));
+                            collapse_link = accordion.GetLinkDiv();
+                            collapse_body = accordion.GetToggleDiv();
+                        }
+                        if(showGlyphs){ // by default add glyphs to a collapsible card
+                        // TODO: Don't use float, just set it to align to the right (need to make icon decend from div first)
+                            prefab::FontAwesomeIcon up("fa-angle-double-up");
+                            prefab::FontAwesomeIcon down("fa-angle-double-down");
+                            card_header << up;
+                            card_header << down;
+                            up.AddClass("float-right btn-link collapse_toggle setting_heading");
+                            down.AddClass("float-right btn-link collapse_toggle setting_heading");
+
+                        }
+                        *this << collapse_link;
+                        *this << collapse_body;
+                        card_header.AddClass("collapse_toggle_card_header");
+                    }
+                    else{ // plain card with no toggle enabled
+                        *this << card_header;
+                        *this << card_body;
```

Comment:
Overall can we refactor this fn to reduce code duplication and do something like make each element separately depending on open/closed/static (header body) and then stream them into the doc at the end? We might also break them out into helper functions make header, make body, etc

Reply:
For collapsible cards, `card_header` and `card_body` will not be explicitly streamed into `this`. They will be wrapped in the CollapseCoupling class and the wrapped versions of each will be streamed with `GetControllerDiv()` and `GetTargetDiv()`